app/components/cases/new-subcase.js

```diff
@@ -44,109 +69,93 @@ export default Component.extend({
-    const latestSubcase = await caze.get('latestSubcase');
-    const date = moment().utc()
-      .toDate();
-    let subcase = await this.createSubcaseObject(caze, date);
-    subcase.set('subcaseName', this.subcaseName);
+    subcase.subcaseName = this.subcaseName;
 
     if (latestSubcase) { // Previous "versions" of this subcase exist
       subcase = await this.copySubcaseProperties(subcase, latestSubcase, fullCopy);
-    } else { // This is a plain new subcase
-      subcase = await subcase.save();
     }
-    await caze.hasMany('subcases').reload();
     return subcase;
-  },
-
-  actions: {
-    closeModal() {
-      this.closeModal();
-    },
-
-    typeChanged(id) {
-      const type = this.store.peekRecord('case-type', id);
-      this.set('showAsRemark', type.get('uri') ===  CONSTANTS.CASE_TYPES.REMARK);
-    },
-
-    toggleIsEditing() {
-      this.toggleProperty('isEditing');
-    },
-
-    async copyFullSubcase() {
-      this.set('isLoading', true);
-      const subcase = await this.copySubcase(true);
-      this.set('item', subcase);
-      this.set('isLoading', false);
-      this.refresh();
-    },
-
-    async createSubCase(event) {
-      event.preventDefault();
-      this.set('isLoading', true);
-      const subcase = await this.copySubcase(false);
-      this.set('item', subcase);
-      this.set('isLoading', false);
-      this.refresh();
-    },
-
-    selectType(type) {
-      this.set('type', type);
-    },
-
-    selectModel(items) {
-      this.set('selectedSubcaseName', items);
-      this.set('subcaseName', items.get('label'));
-    },
-  },
-});
+  }
+
+  @action
+  async copySubcaseProperties(subcase, latestSubcase, fullCopy) {
+    const pieces = await this.loadSubcasePieces(latestSubcase);
+    if (fullCopy) {
+      subcase.linkedPieces = await latestSubcase.linkedPieces;
+      subcase.subcaseName = latestSubcase.subcaseName;
+      subcase.accessLevel = await latestSubcase.accessLevel;
+      subcase.showAsRemark = latestSubcase.showAsRemark;
+      const submissionActivity = this.store.createRecord('submission-activity', {
+        startDate: new Date(),
+        pieces: pieces,
+      });
+      await submissionActivity.save();
+      subcase.submissionActivities.pushObject(submissionActivity);
+    } else {
+      subcase.linkedPieces = pieces;
+    }
+    subcase.mandatees = await latestSubcase.mandatees;
+    subcase.iseCodes = await latestSubcase.iseCodes;
+    subcase.requestedBy = await latestSubcase.requestedBy;
+    return subcase;
+  }
+
+  @task
+  *copyFullSubcase() {
+    const subcase = yield this.createSubcase(true);
+    yield this.args.onSave(
+      {
+        subcase: subcase,
+      });
+  }
+
+  @task
+  *saveSubcase() {
+    const subcase = yield this.createSubcase(false);
+    yield this.args.onSave(
+      {
+        subcase: subcase,
+      });
+  }
+
+  @action
+  selectType(type) {
+    this.type = type;
+  }
+
+  @action
+  selectModel(items) {
```

Comment:
`selectSubcaseName` is een duidelijkere naam voor deze actie.  En argument in enkelvoud want er wordt maar 1 waarde geselecteerd